router/transformer/transformer_test.go

```diff
@@ -0,0 +1,232 @@
+		ExpectedBody:      `Post "%s/v0/destinations/ctx_timeout_dest/proxy": context deadline exceeded`,
+		proxyStatusCode:   http.StatusOK,
+		proxyResponse:     `{"output": {"status": 200, "message": "", "destinationResponse":"ctx_timeout_dest"}}`,
+		proxyTimeoutValue: 4 * time.Millisecond,
+		context: ProxyContextTc{
+			Timeout: 2 * time.Millisecond,
+		},
+		postParametersT: integrations.PostParametersT{
+			Type:          "REST",
+			URL:           "http://www.ctx_timeout_dest.domain.com",
+			RequestMethod: http.MethodPost,
+			QueryParams:   map[string]interface{}{},
+			Body: map[string]interface{}{
+				"JSON": map[string]interface{}{
+					"key_1": "val_1",
+					"key_2": "val_2",
+				},
+				"FORM":       map[string]interface{}{},
+				"JSON_ARRAY": map[string]interface{}{},
+				"XML":        map[string]interface{}{},
+			},
+			Files: map[string]interface{}{},
+		},
+	},
+	"ctx_cancel_dest": {
+		name:            "should throw timeout exception due to context getting cancelled immediately",
+		ExpectedStCode:  http.StatusInternalServerError,
+		ExpectedBody:    `Post "%s/v0/destinations/ctx_cancel_dest/proxy": context canceled`,
+		proxyStatusCode: http.StatusOK,
+		proxyResponse:   `{"output": {"status": 200, "message": "", "destinationResponse":"ctx_cancel_dest"}}`,
+		context: ProxyContextTc{
+			Cancel: true,
+		},
+		postParametersT: integrations.PostParametersT{
+			Type:          "REST",
+			URL:           "http://www.ctx_timeout_dest.domain.com",
+			RequestMethod: http.MethodPost,
+			QueryParams:   map[string]interface{}{},
+			Body: map[string]interface{}{
+				"JSON": map[string]interface{}{
+					"key_1": "val_1",
+					"key_2": "val_2",
+				},
+				"FORM":       map[string]interface{}{},
+				"JSON_ARRAY": map[string]interface{}{},
+				"XML":        map[string]interface{}{},
+			},
+			Files: map[string]interface{}{},
+		},
+	},
+}
+
+func Initialization() {
+	config.Load()
+	stats.Init()
+	stats.Setup()
+	rtTf.Init()
+	logger.Init()
+}
+
+func TestProxyRequest(t *testing.T) {
+	t.Setenv("RSERVER_HTTP_CLIENT_TIMEOUT", "1s")
+	Initialization()
+
+	srvMux := mux.NewRouter()
+	srvMux.HandleFunc("/v0/destinations/{destName}/proxy", proxyHandlerFunc)
+	srv := httptest.NewServer(srvMux)
+	defer srv.Close()
+
+	for destName, tc := range proxyResponseTcs {
+		t.Run(tc.name, func(t *testing.T) {
+			tr := rtTf.NewTransformer()
+			// Logic for executing test-cases not manipulating test-cases
+			if tc.rtTimeout.Milliseconds() > 0 {
+				tr.Setup(tc.rtTimeout)
+			} else {
+				// Just a default value
+				tr.Setup(2 * time.Millisecond)
+			}
+			// Logic to include context timing out
+			ctx := context.TODO()
+			var cancelFunc context.CancelFunc
+			if tc.context.Timeout.Milliseconds() > 0 {
+				ctx, cancelFunc = context.WithTimeout(context.TODO(), tc.context.Timeout)
+				defer cancelFunc()
+			} else if tc.context.Cancel {
+				ctx, cancelFunc = context.WithCancel(context.TODO())
+				cancelFunc()
```

Comment:
Should we use `go cancelFunc()` to mimic it happened concurrently ?

Reply:
Not sure if it makes any noticeable impact. Do you really think, it would help us if we did this ?